### ISSUE TITLE
Extra Nodes in Search Menu

### ISSUE THREAD

**Issue by vicdoval** [nortikin/sverchok#3358]
Right now the sverchok-extra nodes  don't appear on the search menu.

I have been digging into it and if I add 

```
fx.append((str(idx),'Location','',idx))
loop_reverse['Location'] = 'SvExLadyBugLocationNode'
idx +=1
```
into the gather_items function
https://github.com/nortikin/sverchok/blob/74f2e7b46ea4c8a10faf5a2c555d7699f129a9e8/utils/sv_extra_search.py#L100

Then the Location node appears in the search menu and loads succesfully

So it is only needed a function that returns bl_idname and class name from the valid extra nodes to make it work.
But I could not find such a function. `the get_extra_categories` function seems return the categories with a items field but  could not find the way to access them (requires a context?)  
@portnov  am I missing anything?

I think is quite important the nodes to appear in the search menu. Could you please throw some light into this task?

**Comment by portnov** [comment]
As far as I understand, this is a registration order issue. At the moment when gather_items() is executed, it does not contain extra categories. I think you have to reorder the code so that the list of nodes is gathered at the moment the user actually calls the menu first time, not at startup.

**Comment by vicdoval** [comment]
If you run this script from the console (once everything is loaded)
```
from sverchok.utils.extra_categories import get_extra_categories
a = get_extra_categories()
b = a[0].items()
```
The console will return:
`TypeError: items_gen() missing 1 required positional argument: 'context'`
So did not try it to plug it in the gather_items() function....

**Comment by zeffii** [comment]
why not make a new separate extra menu, at least for the time being.  one that includes a category for 'unsorted' nodes. (ie nodes that have been registered in the texteditor..)

**Comment by portnov** [comment]
As far as I understand, this is closed by #3582.
